hub/api/dataset_handler.py

```diff
@@ -0,0 +1,189 @@
+from hub.util.exceptions import DatasetHandlerError, PathNotEmptyException
+from hub.util.storage import get_storage_and_cache_chain
+from typing import Optional
+from hub.constants import DEFAULT_LOCAL_CACHE_SIZE, DEFAULT_MEMORY_CACHE_SIZE, MB
+from .dataset import Dataset
+from hub.util.keys import dataset_exists
+
+
+class dataset:
+    def __new__(
+        cls,
+        path: str,
+        read_only: bool = False,
+        overwrite: bool = False,
+        public: Optional[bool] = True,
+        memory_cache_size: int = DEFAULT_MEMORY_CACHE_SIZE,
+        local_cache_size: int = DEFAULT_LOCAL_CACHE_SIZE,
+        creds: Optional[dict] = None,
+        token: Optional[str] = None,
```

Comment:
```suggestion
        public: Optional[bool] = True,
        memory_cache_size: int = DEFAULT_MEMORY_CACHE_SIZE,
        local_cache_size: int = DEFAULT_LOCAL_CACHE_SIZE,
        creds: dict = None,
        token: str = None,
```

is `public` supposed to be optional?
also, optional isn't necessary when assigning with `None`

Reply:
I think it's good to gave for creds and token, removed for public